x/interchainstaking/keeper/delegation.go

```diff
@@ -413,20 +412,36 @@ func (k *Keeper) MakePerformanceDelegation(ctx sdk.Context, zone *types.Zone, va
 func (k *Keeper) FlushOutstandingDelegations(ctx sdk.Context, zone *types.Zone, delAddrBalance sdk.Coin) error {
 	var amount sdk.Coins
 	exclusionTime := ctx.BlockTime().AddDate(0, 0, -1)
-
-	k.IterateZoneReceipts(ctx, zone, func(_ int64, receiptInfo types.Receipt) (stop bool) {
-		if receiptInfo.FirstSeen.After(exclusionTime) && receiptInfo.Completed == nil {
-			amount = amount.Add(receiptInfo.Amount...)
+	var coins sdk.Coins
+	if ctx.BlockHeight() < 1978500 && ctx.ChainID() == "quicksilver-2" {
+		k.IterateZoneReceipts(ctx, zone, func(_ int64, receiptInfo types.Receipt) (stop bool) {
+			if receiptInfo.FirstSeen.After(exclusionTime) && receiptInfo.Completed == nil {
+				amount = amount.Add(receiptInfo.Amount...)
+			}
+			return false
+		})
+
+		coins = sdk.NewCoins(delAddrBalance).Sub(amount...)
+		if coins.IsAnyNegative() || coins.IsZero() {
+			k.Logger(ctx).Debug("delegate account balance negative, setting outdated reciepts")
+			k.SetReceiptsCompleted(ctx, zone, exclusionTime, ctx.BlockTime())
+			return nil
+		}
+	} else {
+		k.IterateZoneReceipts(ctx, zone, func(_ int64, receiptInfo types.Receipt) (stop bool) {
+			if receiptInfo.FirstSeen.Before(exclusionTime) && receiptInfo.Completed == nil {
+				amount = amount.Add(receiptInfo.Amount...)
+			}
+			return false
+		})
+
+		coins2, hasNeg := sdk.NewCoins(delAddrBalance).SafeSub(amount...)
+		coins = coins2
```

Comment:
```suggestion
		var hasNeg bool
		coins, hasNeg = sdk.NewCoins(delAddrBalance).SafeSub(amount...)
```

Reply:
would be my preference, but simply stylistic (and saves an additional allocation)